### PR TITLE
[daftsex] Add new extractors

### PR DESCRIPTION
**Pull request by k3ns1n**
## Please follow the guide below

- You will be asked some questions, please read them **carefully** and answer honestly
- Put an `x` into all the boxes [ ] relevant to your *pull request* (like that [x])
- Use *Preview* tab to see how your *pull request* will actually look like

---

### Before submitting a *pull request* make sure you have:
- [x] At least skimmed through [contributing guidelines](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#developer-instructions) including [yt-dlp coding conventions](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#yt-dlp-coding-conventions)
- [x] [Searched](https://github.com/yt-dlp/yt-dlp/search?q=is%3Apr&type=Issues) the bugtracker for similar pull requests
- [x] Checked the code with [flake8](https://pypi.python.org/pypi/flake8)

### In order to be accepted and merged into yt-dlp each piece of code must be in public domain or released under [Unlicense](http://unlicense.org/). Check one of the following options:
- [x] I am the original author of this code and I am willing to release it under [Unlicense](http://unlicense.org/)
- [ ] I am not the original author of this code but it is in public domain or released under [Unlicense](http://unlicense.org/) (provide reliable evidence)

### What is the purpose of your *pull request*?
- [ ] Bug fix
- [ ] Improvement
- [x] New extractor
- [ ] New feature

---

### Description of your *pull request* and other information

New extractor

### CODE REVIEW

File: yt_dlp/extractor/daftsex.py [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+from ..utils import (
+    int_or_none,
+    js_to_json,
+    get_elements_by_class,
+    parse_duration,
+    try_get
+)
+
+
+class DAFTSEXIE(InfoExtractor):
```
Comment (Lesmiscore) [inline yt_dlp/extractor/daftsex.py:15]: Only the first character of each word must be upper case.
```suggestion
class DaftsexIE(InfoExtractor):
```

File: yt_dlp/extractor/daftsex.py [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+        video_id = self._match_id(url)
+        webpage = self._download_webpage(url, video_id)
+        title = get_elements_by_class('heading', webpage)[-1]
+        duration = parse_duration(self._search_regex(
+            r'Duration: ([0-9]{2}:[0-9]{2})',
+            webpage, 'duration', fatal=False))
+        views = int_or_none(self._search_regex(
+            r'Views: ([0-9 ]+)',
+            webpage, 'views', fatal=False).replace(' ', ''))
+
```
Comment (pukkandan) [inline yt_dlp/extractor/daftsex.py:36]: utils.parse_count

File: yt_dlp/extractor/daftsex.py [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+        for format_id, format_data in video_params['video']['cdn_files'].items():
+            ext, height = format_id.split('_')
+            extra_quality_data = format_data.split('.')[-1]
+            formats.append({
+                'format_id': format_id,
+                'url': '%s/videos/%s/%s/%s.mp4?extra=%s' % (server_domain,
+                                                            id1,
+                                                            id2,
+                                                            height,
+                                                            extra_quality_data),
```
Comment (pukkandan) [inline yt_dlp/extractor/daftsex.py:71]: use fstring here

File: yt_dlp/extractor/daftsex.py [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+            r'DaxabPlayer\.Init\({[\s\S]*color:\s*"([0-9a-z]+)"[\s\S]*}',
+            webpage, 'player color', fatal=False)
+        if not player_color:
+            player_color = ''
+
+        embed_page = self._download_webpage(
+            'https://daxab.com/player/%s?color=%s' % (player_hash,
+                                                      player_color),
+            video_id,
+            headers={'Referer': url})
```
Comment (pukkandan) [inline yt_dlp/extractor/daftsex.py:50]:
```suggestion
            'https://daxab.com/player/%s?color=%s' % (player_hash, player_color),
            video_id, headers={'Referer': url})
```

File: yt_dlp/extractor/daftsex.py [hunk 5]
```diff
@@ -0,0 +1,88 @@
...
+
+        player_hash = self._search_regex(
+            r'DaxabPlayer\.Init\({[\s\S]*hash:\s*"([0-9a-zA-Z_\-]+)"[\s\S]*}',
+            webpage, 'player hash')
+        player_color = self._search_regex(
+            r'DaxabPlayer\.Init\({[\s\S]*color:\s*"([0-9a-z]+)"[\s\S]*}',
+            webpage, 'player color', fatal=False)
+        if not player_color:
+            player_color = ''
+
```
Comment (pukkandan) [inline yt_dlp/extractor/daftsex.py:45]:
```suggestion
        player_color = self._search_regex(
            r'DaxabPlayer\.Init\({[\s\S]*color:\s*"([0-9a-z]+)"[\s\S]*}',
            webpage, 'player color', fatal=False) or ''

```

**Review by pukkandan — CHANGES_REQUESTED** [review]

File: yt_dlp/extractor/daftsex.py [hunk 6]
```diff
@@ -0,0 +1,82 @@
...
+            'https://daxab.com/player/%s?color=%s' % (player_hash, player_color),
+            video_id, headers={'Referer': url})
+        video_params = self._parse_json(
+            self._search_regex(
+                r'window\.globParams\s*=\s*({[\S\s]+})\s*;\s*<\/script>',
+                embed_page, 'video parameters'),
+            video_id, transform_source=js_to_json)
+
+        server_domain = compat_b64decode(video_params['server'][::-1]).decode('utf-8')
+        server_domain = 'https://' + server_domain
```
Comment (pukkandan) [inline yt_dlp/extractor/daftsex.py:55]:
```suggestion
        server_domain = 'https://%s' % compat_b64decode(video_params['server'][::-1]).decode('utf-8')
```

**Review by pukkandan — CHANGES_REQUESTED** [review]

File: yt_dlp/extractor/daftsex.py [hunk 7]
```diff
@@ -0,0 +1,82 @@
...
+        id1, id2 = video_id.split('_')
+
+        formats = []
+        for format_id, format_data in video_params['video']['cdn_files'].items():
+            ext, height = format_id.split('_')
+            extra_quality_data = format_data.split('.')[-1]
+            url = f'{server_domain}/videos/{id1}/{id2}/{height}.mp4?extra={extra_quality_data}'
```
Comment (pukkandan) [inline yt_dlp/extractor/daftsex.py:62]:
```suggestion
        formats = []
        for format_id, format_data in video_params['video']['cdn_files'].items():
            ext, height = format_id.split('_')
            extra_quality_data = format_data.split('.')[-1]
            url = f'{server_domain}/videos/{video_id.replace("_", "/")}/{height}.mp4?extra={extra_quality_data}'
```

File: yt_dlp/extractor/daftsex.py [hunk 8]
```diff
@@ -0,0 +1,79 @@
...
+
+from .common import InfoExtractor
+from ..compat import compat_b64decode
+from ..utils import (
+    int_or_none,
+    parse_count,
+    js_to_json,
+    get_elements_by_class,
+    parse_duration,
+    try_get
```
Comment (pukkandan) [inline yt_dlp/extractor/daftsex.py:12]: sort

**Comment by Soebb** [comment]
It throws bellow error for some videos (like [this](https://daftsex.com/watch/-145958185_456243473) one)

ERROR:root:ERROR: -145958185_456243473: An extractor error has occurred. (caused by KeyError('cdn_files')); please report this issue on https://github.com/yt-dlp/yt-dlp , filling out the "Broken site" issue template properly. Confirm you are on the latest version using -U

**Pull request merged**